baseband/vdif/frame.py

```diff
@@ -257,13 +252,11 @@ def fromfile(cls, fh, thread_ids=None, sort=True, edv=None, verify=True):
         if thread_ids and len(frames) < len(thread_ids):
             raise IOError("could not find all requested frames.")
 
-        if sort:
-            # Sort by thread id, by user order if needed.
-            if thread_ids:
-                frames.sort(key=lambda frame:
-                            thread_ids.index(frame['thread_id']))
-            else:
-                frames.sort(key=lambda frame: frame['thread_id'])
+        # Turn dict of frames into a list, following order given by
+        # thread_ids, or just sorting by their own thread_id
+        if thread_ids is None:
+            thread_ids = sorted(frames.keys())
+        frames = [frames[thread_id] for thread_id in thread_ids]
```

Comment:
The linter in my text editor complains that `thread_id` is modified by this line.  Not sure if it's best practice to create a unique dummy variable for list comprehension.